### PR TITLE
Criteria for doing API reviews and design reviews in TF

### PR DESCRIPTION
**Pull request by alextp**
These documents show what we mostly look for in API reviews and design reviews in general. They should be edited in the future as our practices change.

### CODE REVIEW

**Comment by ewilderj** [comment]
This may need a little extra elaboration to explain terms to non-TensorFlow team members . Tagging in @theadactyl for her opinion.

File: governance/api-reviews.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+don’t-really-want-but-need-for-now; experimental APIs should be APIs that we
+intend to make standard in the near future, but have some lingering questions to
+resolve first.
+
+
+### Docstrings 
+
+TF APIs should have comprehensive documentation in the form of docstrings. If at
+all possible these docstrings should have runnable examples, and these examples
+should form a doctest so they stay correct. The examples should demonstrate an
```
Comment (MarkDaoust) [inline governance/api-reviews.md]: Can we link to this doc which describes it/gives more examples:

https://www.tensorflow.org/community/contribute/docs_ref

Comment (alextp) [reply governance/api-reviews.md]: Done

File: governance/api-reviews.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+We prefer experimental namespaces (i.e. tf.data.experimental.foobar) over
+experimental-prefixed names (i.e. tf.data.experimental_foobar) except when
+adding an experimental class method, or an experimental argument. Experimental
+endpoints should be deprecated in a minor release before they can be removed in
+the next. We would like new experimental symbols to be things which will
+eventually end up in core TF as opposed to things we expect will be phased out
+with no clear replacement. The best expectation to have for an experimental
+endpoint is that the “experimental” will simply be removed. If you don’t believe
+that’ll work, it should probably not be added in its current form.  Style
```
Comment (mihaimaruseac) [inline governance/api-reviews.md]: Should `Style` here be a different section header?

Comment (alextp) [reply governance/api-reviews.md]: Oops.

File: governance/api-reviews.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+that new arguments are added after the last existing argument, even if that
+results in name not being the last argument.
+
+We generally prefer spelling things out over using abbreviations except when
+abbreviations are more standard than spelling things out (i.e. don’t spell out
+linalg or svd). When in doubt we ask domain experts or use web search to see
+what spelling is most common.
+
+If possible we prefer to name things in a similar way to numpy (e.g., we would
+not pick einsum as a name, but numpy and MatLab before it has, and that
```
Comment (shoyer) [inline governance/api-reviews.md]: nit: I'm pretty sure einsum is not an official matlab function. I only see a version on MathWork's FileExchange website: https://www.mathworks.com/matlabcentral/fileexchange/68995-einsum

File: governance/api-reviews.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+
+## Keras changes
+
+Keras has its own external API and maintainers, as well as an internal
+team. Keras API changes tend to be cross-cutting as core Keras classes are base
+classes to many TF API endpoints. We generally only look at keras changes after
+a keras owner (say, fchollet or karmel) has already approved it on their
+end. fchollet has guidelines for changes to the Keras API at
+go/keras-api-guidelines. These include many generally good practices that are
```
Comment (shoyer) [inline governance/api-reviews.md]: This link should be updated for non-Googlers :)

Comment (bhack) [reply governance/api-reviews.md]: Where this will be documented as Keras will go on a separted org repo https://github.com/tensorflow/community/pull/202?

**Pull request merged**